File: internal/websocket/message.go
package websocket

import "time"

const (
	chatType         = "Chat"
	notificationType = "Notification"
)

type Message struct {
	Type      string `json:"type"`
	UserID    *int   `json:"userId,omitempty"`
	Username  string `json:"username,omitempty"`
	Content   string `json:"content"`
	Timestamp string `json:"timestamp"`
	Color     string `json:"color"`
}

func NewChatMessage(user *User, typeMsg, content string) Message {
	return Message{
		Type:      typeMsg,
		UserID:    &user.ID,
		Username:  user.Username,
		Content:   content,
		Timestamp: time.Now().Format(time.RFC3339),
		Color:     user.Color,
	}
}

func NewNotificationMessage(content string) Message {
	return Message{
		Type:      notificationType,
		Content:   content,
		Timestamp: time.Now().Format(time.RFC3339),
		Color:     "#666666",
	}
}

File: internal/frontend/olha-mensagem-app/src/lib/types/message.ts
import type { ChatMessage } from './websocket.types';

export type MessageDto = {
	id: number;
	channelId: number;
	userId: number;
	userUsername: string;
	userColor: string;
	content: string;
	timestamp: string;
};

export function messageDtoToChatMessage(message: MessageDto): ChatMessage {
	return {
		type: 'Chat',
		userId: message.userId,
		username: message.userUsername,
		content: message.content,
		timestamp: message.timestamp,
		color: message.userColor
	};
}
